### PR TITLE
replay add stopheight

### PR DESCRIPTION
**Pull request by tianxiataiping**
- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/okex/okexchain/blob/develop/CONTRIBUTING.md#pr-targeting))

- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/okex/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: cmd/exchaind/replay.go [hunk 1]
```diff
@@ -161,15 +165,40 @@ func initChain(state sm.State, stateDB dbm.DB, genDoc *types.GenesisDoc, proxyAp
 	return nil
 }
 
+//func doReplay(ctx *server.Context, state sm.State, stateStoreDB dbm.DB,
```
Comment (KamiD) [inline cmd/exchaind/replay.go]: delete those annotated code

File: cmd/exchaind/replay.go [hunk 2]
```diff
@@ -161,15 +165,40 @@ func initChain(state sm.State, stateDB dbm.DB, genDoc *types.GenesisDoc, proxyAp
...
 	originBlockStore := store.NewBlockStore(originBlockStoreDB)
 	originLatestBlockHeight := originBlockStore.Height()
 	log.Println("origin latest block height", "height", originLatestBlockHeight)
 
-	for height := startBlockHeight; height <= originLatestBlockHeight; height++ {
+	stopheight := stopBlockHeight
+	if stopheight == 0 {
+		stopheight = originLatestBlockHeight
+	}
+	log.Println("replay stop block height", "height", stopheight)
```
Comment (KamiD) [inline cmd/exchaind/replay.go]: add check of stopheight < startBlockHeight

File: cmd/exchaind/replay.go [hunk 3]
```diff
@@ -29,6 +29,7 @@ const (
 	blockStoreDB  = "blockstore"
 	stateDB       = "state"
 	pprofAddrFlag = "pprof_addr"
+	FlagHaltHeight = "height"
```
Comment (xiangjianmeng) [inline cmd/exchaind/replay.go]: FlagHaltHeight has been defined, not need to define a new

File: cmd/exchaind/replay.go (hunk 2 above)
Comment (ylsGit) [reply cmd/exchaind/replay.go]: min(stopheight ,startBlockHeight)

**Review by xiangjianmeng — APPROVED** [review]

**Pull request merged**